File: NET/JNetBase.Sharp/Threads/ThreadExecutor.cs
using System.Collections.Generic;
using System.Threading;

namespace JNetBase.Sharp.Threads
{
    public sealed class ThreadExecutor : IExecutor
    {
        private readonly IList<Thread> _threads;

        public ThreadExecutor()
        {
            _threads = new List<Thread>();
        }

        public void Dispose()
        {
            foreach (var thread in _threads)
                thread.Interrupt();
            _threads.Clear();
        }

        public Thread CreateThread(ThreadStart action, string name)
        {
            var task = new Thread(action)
            {
                IsBackground = true
            };
            if (name != null)
                task.Name = name;
            _threads.Add(task);
            task.Start();
            return task;
        }
    }
}

File: JVM/jnetbase-java/src/main/java/jnetbase/java/threads/ThreadExecutor.java
package jnetbase.java.threads;

import java.util.LinkedList;
import java.util.List;

public final class ThreadExecutor implements IExecutor {

    private final List<Thread> _threads;

    public ThreadExecutor() {
        _threads = new LinkedList<Thread>();
    }

    @Override
    public void close() {
        for (Thread thread : _threads)
            thread.interrupt();
        _threads.clear();
    }

    @Override
    public Thread createThread(Runnable action, String name) {      	
        Thread task = new Thread(action);
        task.setDaemon(true);
        if (name != null)
          	task.setName(name);
        _threads.add(task);
        task.start();
        return task;
    }

    @Override
    public Thread newThread(Runnable action) {
        return createThread(action, null);
    }
}
